tools/stringparser/main.c

```diff
@@ -57,9 +57,16 @@ int main(int argc, char* argv[]) {
     const char *str = *astr->sval;
 
     if (strcmp("-", *astr->sval) == 0) {
-        from_stdin = 1;
-        char *tmp_str = malloc(512);
-        fgets(tmp_str, 512, stdin);
+        if (fgets(tmp_str, 512, stdin) == NULL) {
+            if (feof(stdin)) {
+                printf("Unexpected end of stdin\n");
+            } else if (ferror(stdin)) {
+                printf("Error reading stdin\n");
```

Comment:
Here it is unclear how to find out what the error was, since man page explicitly says these functions do not set errno..